### PR TITLE
🗑 refactor(loss, nn, recipes, transforms, optim): remove unused local variables

### PR DESCRIPTION
**Pull request by Horgix**

### CODE REVIEW

File: torchelie/nn/layers.py [hunk 1]
```diff
@@ -92,7 +92,6 @@ def forward(self, x: torch.Tensor) -> torch.Tensor:
         v = self.value(x_flat)
 
         affinity = torch.einsum('bki,bkj->bij', q, k)
-        attention = F.softmax(affinity, dim=1)
```
Comment (Vermeille) [inline torchelie/nn/layers.py:95]: This is a bug indeed. The einsum() below should use attention rather than affinity

File: torchelie/loss/neuralstyleloss.py [hunk 2]
```diff
@@ -14,8 +14,6 @@
 
 @tu.experimental
 def hist_match(source, template):
-    oldshape = source.shape
-
```
Comment (Vermeille) [inline torchelie/loss/neuralstyleloss.py:18]: thx

File: torchelie/loss/neuralstyleloss.py [hunk 3]
```diff
@@ -96,10 +94,6 @@ def get_style_content_(self, img: torch.Tensor,
                            detach: bool) -> Dict[str, Dict[str, torch.Tensor]]:
         activations: Dict[str, torch.Tensor]
         _, activations = self.net(self.norm(img), detach=detach)
-        style = {
-            l: a
-            for l, a in activations.items() if l in self.style_layers
-        }
```
Comment (Vermeille) [inline torchelie/loss/neuralstyleloss.py:102]: good catch.

File: torchelie/nn/functional/vq.py [hunk 4]
```diff
@@ -47,7 +47,7 @@ def backward(ctx, straight_through, unused_indices):
         codes, inputs, beta, codebook, indices = ctx.saved_tensors
 
         # TODO: figure out proper vq loss reduction
-        vq_loss = F.mse_loss(inputs, codes).detach()
```
Comment (Vermeille) [inline torchelie/nn/functional/vq.py:50]: good catch

File: torchelie/nn/resblock.py [hunk 5]
```diff
@@ -69,7 +69,6 @@ def __init__(self,
 
         self.pre = CondSeq()
 
-        mid = out_channels // 4
```
Comment (Vermeille) [inline torchelie/nn/resblock.py:72]: thx

File: torchelie/optim.py [hunk 6]
```diff
@@ -40,8 +40,6 @@ def step(self, closure=None):
                 # Perform stepweight decay
                 p.data.mul_(1 - group['lr'] * group['weight_decay'])
 
-                state = self.state[p]
-
```
Comment (Vermeille) [inline torchelie/optim.py:44]: I don't feel quite good about removing this one actually. Commenting this line would feel better. I don't want to forget that there is an inherited `state` dict that could be extremely helpful.

File: torchelie/recipes/upsamplinggan.py [hunk 7]
```diff
@@ -192,7 +192,6 @@ class PerceptualDiscriminator(nn.Module):
     @tu.experimental
     def __init__(self, n_downscales=4):
         super().__init__()
-        layers = tu.FrozenModule(tch.models.PerceptualNet(layers))
```
Comment (Vermeille) [inline torchelie/recipes/upsamplinggan.py:195]: Disregard everything in this file.

File: torchelie/transforms/__init__.py [hunk 8]
```diff
@@ -175,7 +175,6 @@ def get_params(img, scale, ratio=1):
         """
         scale = math.sqrt(scale)
         width, height = ResizedCrop._get_image_size(img)
-        area = height * width
```
Comment (Vermeille) [inline torchelie/transforms/__init__.py:178]: good catch, copy-paste remnants

File: torchelie/nn/functional/vq.py (hunk 4 above)
Comment (Vermeille) [reply torchelie/nn/functional/vq.py:50]: Commenting it is definitely the right thing to do

**Review by Vermeille — CHANGES_REQUESTED** [review]
Thank you so much!
Marking it as "Request change" so that I don't forget about the SelfAttention2d thingy

**Review by Vermeille — COMMENTED** [review]
All good!

File: torchelie/nn/layers.py (hunk 1 above)
Comment (Horgix) [reply torchelie/nn/layers.py:95]: Opened issue #10 to make sure to not forget about it! I guess it might make sense to remove the dead variable in the meantime since the issue have a link to a commit that still has the line?

File: torchelie/optim.py (hunk 6 above)
Comment (Horgix) [reply torchelie/optim.py:44]: In which way could it be extremely helpful? Could a comment about that as a reminder instead of an unused variable be more fitting? :slightly_smiling_face:

File: torchelie/recipes/upsamplinggan.py (hunk 7 above)
Comment (Horgix) [reply torchelie/recipes/upsamplinggan.py:195]: Should I take it as a request to keep this line in and revert this change from the PR?

File: torchelie/nn/layers.py (hunk 1 above)
Comment (Horgix) [reply torchelie/nn/layers.py:95]: Solved by 02a9f2f73848e48381e3b824f62c7006b3a9eab2
